### PR TITLE
Fix constraints about mirage and docteur-{unix,solo5}

### PR DESCRIPTION
**Pull request by dinosaure**
Related to #20621

### CODE REVIEW

File: packages/docteur-unix/docteur-unix.0.0.2/opam [hunk 1]
```diff
@@ -11,7 +11,7 @@ depends: [
   "ocaml" {>= "4.07.0"}
   "dune" {>= "2.8.0"}
   "docteur" {= version}
-  "mirage-unix" {>= "4.0.0"}
+  "mirage-unix" {>= "4.0.0" & < "5.0.0"}
```
Comment (samoht) [inline packages/docteur-unix/docteur-unix.0.0.2/opam:14]: I still don't understand why we need an upper bound here. What error is this fixing?

Comment (samoht) [reply packages/docteur-unix/docteur-unix.0.0.2/opam:14]: ok got it, mirage3 needs mirage-unix.4 and mirag4 needs mirage-unix.5

**Comment by hannesm** [comment]
my impression from https://github.com/ocaml/opam-repository/pull/20621 is that docteur and docteur-solo5 use "OS.Solo5", and their `mirage-solo5` dependency needs an upper bound (i.e. the docteur-solo5 fix here is fine, but docteur still needs an upper bound, and I can't see anything about docteur-unix).

**Comment by dinosaure** [comment]
Again, all of these constraints is due to our rename of `OS` to `<target>_os`. I will cut a release of `docteur` after the release of `mirage`. It's fine to merge this PR and lint our world between MirageOS 3 and MirageOS 4.

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**